src/app/src/pages/Draw.js

```diff
@@ -1,65 +1,18 @@
-import { useSelector } from 'react-redux';
-import { Box, Flex, Spinner } from '@chakra-ui/react';
+import { Box, Flex } from '@chakra-ui/react';
 
-import NotFound from './NotFound';
 import DrawTools from '../components/DrawTools';
 import Layers from '../components/Layers';
 import Map from '../components/Map';
 import Sidebar from '../components/Sidebar';
 
-import { useGetBoundaryDetailsQuery } from '../api/boundaries';
-import { BOUNDARY_STATUS, ROLES } from '../constants';
-import { useBoundaryId } from '../hooks';
-
-const DRAW_MODES = {
-    FULLY_EDITABLE: 'fully_editable',
-    ANNOTATIONS_ONLY: 'annotations_only',
-    READ_ONLY: 'read_only',
-};
-
 export default function Draw() {
-    const user = useSelector(state => state.auth.user);
-    const id = useBoundaryId();
-
-    const { isFetching, data: details, error } = useGetBoundaryDetailsQuery(id);
-
-    if (isFetching) {
-        return (
-            <Box w='100%' h='100vh'>
-                <Flex direction='column' alignItems='center'>
-                    <Spinner mt={60} />
-                </Flex>
-            </Box>
-        );
-    }
-
-    if (error || typeof details !== 'object') {
-        return <NotFound />;
-    }
-
-    let mode = DRAW_MODES.READ_ONLY;
-
-    if (
-        [BOUNDARY_STATUS.SUBMITTED, BOUNDARY_STATUS.IN_REVIEW].includes(
-            details.status
-        ) &&
-        user.role === ROLES.VALIDATOR
-    ) {
-        mode = DRAW_MODES.ANNOTATIONS_ONLY;
-    } else if (
-        details.status === BOUNDARY_STATUS.DRAFT &&
-        user.role === ROLES.CONTRIBUTOR
-    ) {
-        mode = DRAW_MODES.FULLY_EDITABLE;
-    }
-
     return (
         <Flex>
             <Sidebar />
             <Box flex={1} position='relative'>
                 <Map>
-                    <Layers mode={mode} />
```

Comment:
Is `Layers` going to need access to `mode`? If so, maybe boundary details should be loaded in a separate component above `DrawTools` and `Layers`, but below `Map`.

Reply:
I like that arrangement.

Alternatively, if `Layers` uses the same RTKQuery hook to get the boundary details, will it not get a cached value? i.e. it should be fetched only once for the different components that request it, right?

Reply:
> Alternatively, if Layers uses the same RTKQuery hook to get the boundary details, will it not get a cached value? i.e. it should be fetched only once for the different components that request it, right?

Yep, but then we would also need to handle the various states of the request (isFetching, error, success) in that component as well. I like to have one parent component handling all this and passing the non-null data to the children.